src/llm/apis/openai_completions.cpp

```diff
@@ -128,6 +131,45 @@ ov::Tensor load_image_stbi(const std::string& imageBytes) {
         SharedImageAllocator{image, desiredChannels, y, x});
 }
 
+static size_t appendChunkCallback(void* downloadedChunk, size_t size, size_t nmemb,
+    void* image) {
+    size_t realsize = size * nmemb;
+    auto& mem = *static_cast<std::string*>(image);
+    mem.append(static_cast<char*>(downloadedChunk), realsize);
+    return realsize;
+}
+#define CURL_SETOPT(setopt) \
+    if (status == CURLE_OK) { \
+        status = setopt; \
+    }
+
+static absl::Status downloadImage(const char* url, std::string& image, const double& sizeLimit) {
+    CURL* curl_handle = curl_easy_init();
+    auto status = curl_easy_setopt(curl_handle, CURLOPT_URL, url);
+    CURL_SETOPT(curl_easy_setopt(curl_handle, CURLOPT_WRITEFUNCTION, appendChunkCallback))
+    CURL_SETOPT(curl_easy_setopt(curl_handle, CURLOPT_WRITEDATA, &image))
+    CURL_SETOPT(curl_easy_setopt(curl_handle, CURLOPT_SSL_OPTIONS, CURLSSLOPT_NATIVE_CA))
+    CURL_SETOPT(curl_easy_setopt(curl_handle, CURLOPT_TIMEOUT, 20L))
+    CURL_SETOPT(curl_easy_setopt(curl_handle, CURLOPT_FOLLOWLOCATION, 1L))
+    CURL_SETOPT(curl_easy_setopt(curl_handle, CURLOPT_HTTPPROXYTUNNEL, 1L))
+    CURL_SETOPT(curl_easy_setopt(curl_handle, CURLOPT_MAXFILESIZE, sizeLimit))
+
+    if (status == CURLE_OK) {
+        status = curl_easy_perform(curl_handle);
+    }
```

Comment:
I would consider adding some debug logging. If not on per-opt basis in the macro, then at least something like:
"Image download failed. Could not set all required cURL opts."